### ISSUE TITLE
What kind of tricks should I use to improve performance for near-infared detecion

### ISSUE THREAD

**Issue by JasonSloan** [ultralytics/yolov5#13575]
### Search before asking

- [x] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and [discussions](https://github.com/ultralytics/yolov5/discussions) and found no similar questions.

### Question

Hi team,

I'm currently training a person detection model using YOLOv11 on a dataset consisting of near-infrared (NIR) images. However, I'm facing some performance issues — particularly low recall — which I suspect are due to the low contrast and unique characteristics of NIR images (e.g., subtle texture, dim lighting).

![Image](https://github.com/user-attachments/assets/cee2683c-332b-4639-b469-b77b54a8923f)

Could you please advise on the following:

Are there any recommended training strategies, augmentations, or preprocessing steps when working with NIR imagery?

Would you suggest any specific model configurations or architecture tweaks that help in low-contrast scenarios?

Is there support or guidance for domain adaptation or fine-tuning with limited NIR data?

Are there known limitations or best practices for YOLOv11 in non-visible spectrum (e.g., NIR) conditions?

Thanks for your great work and looking forward to your suggestions!

Best regards.

**Comment by UltralyticsAssistant** [comment]
👋 Hello @JasonSloan, thank you for your interest in YOLOv5 🚀! Please visit our ⭐️ [Tutorials](https://docs.ultralytics.com/yolov5/) for comprehensive guides, including [Custom Data Training](https://docs.ultralytics.com/yolov5/tutorials/train_custom_data/) and advanced strategies like [Hyperparameter Evolution](https://docs.ultralytics.com/yolov5/tutorials/hyperparameter_evolution/).

If this is a 🐛 Bug Report, please provide a **minimum reproducible example** to help us debug it.

If this is a custom training ❓ Question (such as your inquiry about NIR imagery and performance improvements), please provide as much detail as possible, including example images, training logs, and verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

## Requirements

[**Python>=3.8.0**](https://www.python.org/) with all [requirements.txt](https://github.com/ultralytics/yolov5/blob/master/requirements.txt) installed including [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/). To get started:
```bash
git clone https://github.com/ultralytics/yolov5  # clone
cd yolov5
pip install -r requirements.txt  # install
```

## Environments

YOLOv5 can be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://bit.ly/yolov5-paperspace-notebook"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"></a> <a href="https://colab.research.google.com/github/ultralytics/yolov5/blob/master/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolov5"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/yolov5"><img src="https://img.shields.io/docker/pulls/ultralytics/yolov5?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/yolov5/actions/workflows/ci-testing.yml"><img src="https://github.com/ultralytics/yolov5/actions/workflows/ci-testing.yml/badge.svg" alt="YOLOv5 CI"></a>

If this badge is green, all [YOLOv5 GitHub Actions](https://github.com/ultralytics/yolov5/actions) Continuous Integration (CI) tests are currently passing. CI tests verify correct operation of YOLOv5 [training](https://github.com/ultralytics/yolov5/blob/master/train.py), [validation](https://github.com/ultralytics/yolov5/blob/master/val.py), [inference](https://github.com/ultralytics/yolov5/blob/master/detect.py), [export](https://github.com/ultralytics/yolov5/blob/master/export.py) and [benchmarks](https://github.com/ultralytics/yolov5/blob/master/benchmarks.py) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

---

This is an automated response 🤖. An Ultralytics engineer will review your question and assist you soon!

**Comment by pderrenger** [comment]
Hi @JasonSloan,

Thanks for your question about NIR detection. Data augmentation is key for improving performance in this domain. I recommend:

1. Implement Albumentations augmentation integration with YOLO11, focusing on CLAHE (Contrast Limited Adaptive Histogram Equalization) which specifically boosts contrast in dark/low-visibility areas - perfect for NIR imagery.

2. Add augmentations like blur, median blur, and grayscale to help the model learn from varying image qualities, making it more robust for NIR conditions.

3. Consider multi-stage training: first on visible spectrum images (transfer learning), then fine-tune with your NIR dataset.

4. Experiment with a higher learning rate and smaller batch size for domain-specific adaptation.

5. For low-contrast scenarios, try increasing the confidence threshold during inference to reduce false positives.

The [Albumentations integration](https://docs.ultralytics.com/integrations/albumentations/) documentation can help you implement these augmentations effectively. Let me know if you need more specific guidance!

**Comment by JasonSloan** [comment]
> Hi [@JasonSloan](https://github.com/JasonSloan),
> 
> Thanks for your question about NIR detection. Data augmentation is key for improving performance in this domain. I recommend:
> 
> 1. Implement Albumentations augmentation integration with YOLO11, focusing on CLAHE (Contrast Limited Adaptive Histogram Equalization) which specifically boosts contrast in dark/low-visibility areas - perfect for NIR imagery.
> 2. Add augmentations like blur, median blur, and grayscale to help the model learn from varying image qualities, making it more robust for NIR conditions.
> 3. Consider multi-stage training: first on visible spectrum images (transfer learning), then fine-tune with your NIR dataset.
> 4. Experiment with a higher learning rate and smaller batch size for domain-specific adaptation.
> 5. For low-contrast scenarios, try increasing the confidence threshold during inference to reduce false positives.
> 
> The [Albumentations integration](https://docs.ultralytics.com/integrations/albumentations/) documentation can help you implement these augmentations effectively. Let me know if you need more specific guidance!


Hi [pderrenger](https://github.com/pderrenger),

Thank you for your detailed and helpful suggestions regarding NIR detection with YOLOv11. The recommendation to use CLAHE for contrast enhancement in low-visibility NIR imagery is particularly insightful.

To optimize its implementation, could you clarify whether CLAHE should be applied:

​​During training​​ (as part of augmentations),
​​During inference​​ (preprocessing input frames), or
​​Both stages​​ for consistency?
Additionally, are there any trade-offs (e.g., overfitting to enhanced contrasts if used only in training, or latency if applied during inference)?
If you’re aware of any open-source NIR person detection datasets (e.g., for benchmarking or transfer learning), I’d be grateful for pointers.

Thanks again for your guidance—I’ll proceed with Albumentations integration while awaiting your advice on this nuance.

Best regards,
Jason Sloan

**Comment by pderrenger** [comment]
Hi @JasonSloan,

Great questions about implementing CLAHE effectively:

For NIR imagery, I recommend applying CLAHE:
- During training as part of your augmentation pipeline to help the model learn contrast variations
- For inference preprocessing on your actual deployment data, especially if the production environment has similar low-contrast characteristics

The main trade-offs include:
- Training-only application may cause domain shift during inference
- Inference-only application might present features the model wasn't trained to recognize
- Runtime latency impact during inference is typically minimal with optimized implementations

Regarding NIR datasets, consider:
- FLIR Thermal Dataset (contains person annotations in thermal/IR spectrum)
- KAIST Multispectral Pedestrian Dataset (includes aligned visible and thermal IR images)
- Microsoft Research RGB-NIR Scene Dataset (for general domain adaptation)

The Albumentations documentation shows how to create a consistent pipeline that can be applied both during training and inference. This consistency is key for robust NIR detection.

Paula

**Comment by JasonSloan** [comment]
> Hi [@JasonSloan](https://github.com/JasonSloan),
> 
> Great questions about implementing CLAHE effectively:
> 
> For NIR imagery, I recommend applying CLAHE:
> 
> * During training as part of your augmentation pipeline to help the model learn contrast variations
> * For inference preprocessing on your actual deployment data, especially if the production environment has similar low-contrast characteristics
> 
> The main trade-offs include:
> 
> * Training-only application may cause domain shift during inference
> * Inference-only application might present features the model wasn't trained to recognize
> * Runtime latency impact during inference is typically minimal with optimized implementations
> 
> Regarding NIR datasets, consider:
> 
> * FLIR Thermal Dataset (contains person annotations in thermal/IR spectrum)
> * KAIST Multispectral Pedestrian Dataset (includes aligned visible and thermal IR images)
> * Microsoft Research RGB-NIR Scene Dataset (for general domain adaptation)
> 
> The Albumentations documentation shows how to create a consistent pipeline that can be applied both during training and inference. This consistency is key for robust NIR detection.
> 
> Paula

Hi Paula,

Thank you for the detailed and helpful recommendations regarding CLAHE implementation and dataset suggestions. Your point about maintaining consistency between training and inference pipelines is particularly insightful.

I reviewed the FLIR and KAIST datasets you mentioned, but noticed they appear to be thermal infrared (8-14μm) rather than near-infrared (NIR, 700-1000nm) imagery. Since my project specifically focuses on NIR spectrum detection:

Do you know if these thermal datasets would still provide meaningful transfer learning benefits for NIR person detection, given the different wavelength characteristics?
Are you aware of any dedicated NIR person detection datasets similar to these?
I'll proceed with implementing CLAHE in both phases as suggested, but wanted to confirm the dataset applicability before incorporating them. Thanks again for your expertise!

Best regards,
Jason Sloan

**Comment by pderrenger** [comment]
Hi Jason,

You're absolutely right about the wavelength distinction - thermal IR (8-14μm) and NIR (700-1000nm) capture significantly different physical properties. While thermal IR detects heat signatures, NIR primarily captures reflected light beyond visible spectrum.

Regarding transfer learning between these domains:
- Thermal IR datasets would provide limited benefit for NIR detection due to their fundamentally different visual characteristics
- The feature representations learned from thermal data (heat patterns) don't translate well to NIR's reflected light patterns

For NIR-specific datasets, consider:
- SceneNet RGB-D (contains some NIR imagery components)
- EPFL BIWI Kinect NIR Face Dataset (primarily faces but useful for NIR domain understanding)
- NII Shonan NIR datasets (if available)

Alternatively, you might consider creating a synthetic NIR dataset by applying NIR simulation filters to RGB datasets with person annotations, which can serve as pre-training before fine-tuning on your actual NIR data.

You might also experiment with domain adaptation techniques specifically designed to bridge the visible-to-NIR gap rather than thermal-to-NIR.

Paula

**Comment by JasonSloan** [comment]
> Hi Jason,
> 
> You're absolutely right about the wavelength distinction - thermal IR (8-14μm) and NIR (700-1000nm) capture significantly different physical properties. While thermal IR detects heat signatures, NIR primarily captures reflected light beyond visible spectrum.
> 
> Regarding transfer learning between these domains:
> 
> * Thermal IR datasets would provide limited benefit for NIR detection due to their fundamentally different visual characteristics
> * The feature representations learned from thermal data (heat patterns) don't translate well to NIR's reflected light patterns
> 
> For NIR-specific datasets, consider:
> 
> * SceneNet RGB-D (contains some NIR imagery components)
> * EPFL BIWI Kinect NIR Face Dataset (primarily faces but useful for NIR domain understanding)
> * NII Shonan NIR datasets (if available)
> 
> Alternatively, you might consider creating a synthetic NIR dataset by applying NIR simulation filters to RGB datasets with person annotations, which can serve as pre-training before fine-tuning on your actual NIR data.
> 
> You might also experiment with domain adaptation techniques specifically designed to bridge the visible-to-NIR gap rather than thermal-to-NIR.
> 
> Paula

Hi Paula,

Thank you for the clear explanation regarding the thermal/NIR distinction and for the NIR-specific dataset recommendations - this helps narrow my search significantly. Your point about creating synthetic NIR data is particularly interesting, as it could help address my current data limitations.

Would you happen to know of any existing repositories or tools that specialize in:

Realistic RGB-to-NIR image translation (e.g., physics-based spectral simulation)
NIR-style data augmentation pipelines
Domain adaptation frameworks for visible-to-NIR transfer?
I'm aware of general image translation tools like CycleGAN, but wonder if you've encountered any solutions specifically optimized for NIR characteristics. Any pointers would be greatly appreciated as I explore this synthetic data approach.

Thanks again for your invaluable guidance throughout this process!

Best regards,
Jason Sloan

**Comment by pderrenger** [comment]
Hi Jason,

For RGB-to-NIR translation and domain adaptation, here are some specific resources:

For synthetic NIR generation:
- NIRTrans repository (uses GAN-based approaches specifically for visible-to-NIR translation)
- Pix2Pix or CycleGAN frameworks with specialized NIR training pairs
- NVIDIA's Vid2Vid framework (can be adapted for spectral translation with proper training data)

For NIR-specific augmentation:
- Beyond standard Albumentations, consider implementing custom NIR noise models that simulate sensor-specific characteristics
- The Python spectral library offers specific NIR band simulation capabilities

For domain adaptation frameworks:
- ADDA (Adversarial Discriminative Domain Adaptation) has shown promise for cross-spectral tasks
- UNIT (UNsupervised Image-to-image Translation) can be effective for NIR adaptation
- MMDetection's cross-domain detection modules can be adapted for visible-to-NIR transfer

Many researchers also use simple techniques like histogram matching between visible and NIR image pairs as a starting point before more advanced approaches.

Paula

**Comment by JasonSloan** [comment]
> Hi Jason,
> 
> For RGB-to-NIR translation and domain adaptation, here are some specific resources:
> 
> For synthetic NIR generation:
> 
> * NIRTrans repository (uses GAN-based approaches specifically for visible-to-NIR translation)
> * Pix2Pix or CycleGAN frameworks with specialized NIR training pairs
> * NVIDIA's Vid2Vid framework (can be adapted for spectral translation with proper training data)
> 
> For NIR-specific augmentation:
> 
> * Beyond standard Albumentations, consider implementing custom NIR noise models that simulate sensor-specific characteristics
> * The Python spectral library offers specific NIR band simulation capabilities
> 
> For domain adaptation frameworks:
> 
> * ADDA (Adversarial Discriminative Domain Adaptation) has shown promise for cross-spectral tasks
> * UNIT (UNsupervised Image-to-image Translation) can be effective for NIR adaptation
> * MMDetection's cross-domain detection modules can be adapted for visible-to-NIR transfer
> 
> Many researchers also use simple techniques like histogram matching between visible and NIR image pairs as a starting point before more advanced approaches.
> 
> Paula

Hi Paula,

Thank you for the incredibly detailed recommendations—this is exactly the type of targeted guidance I needed! The breakdown across generation tools (NIRTrans, Vid2Vid), augmentation approaches (spectral library), and domain adaptation frameworks (ADDA, UNIT) gives me a clear roadmap to explore.

I'll start by implementing these solutions systematically, beginning with the GAN-based translation approaches you suggested. The histogram matching tip is particularly appealing as a quick baseline before diving into more complex methods.

Should I encounter any implementation challenges or discover interesting results during testing, may I follow up with you? Your expertise has been invaluable in navigating this cross-spectral detection problem.

Grateful for your generosity with these recommendations—I'll be sure to share any meaningful findings from my experiments.

Best regards,
Jason

**Comment by pderrenger** [comment]
Hi Jason,

You're welcome! I'm glad these recommendations provide a clear path forward for your NIR detection project. The systematic approach you're taking is excellent - starting with simpler methods like histogram matching before moving to more complex approaches makes perfect sense.

Feel free to follow up with any implementation challenges or interesting results you discover. Our GitHub issues or the [Ultralytics Community forum](https://community.ultralytics.com/) are great places to share your findings and get additional feedback from our team and other users working on similar problems.

Your work on NIR detection could be valuable to others in the community facing similar cross-spectral challenges.
